### PR TITLE
[core] Better Actor Representation

### PR DESCRIPTION
**Pull request by richardliaw**
This is a shot at a better string representation for actors. 

Right now, the actor handle returned does not provide information about what class the actor handle is wrapping.

Suggestions and improvements welcome.

cc @stephanie-wang @robertnishihara @pcmoritz

### CODE REVIEW

File: python/ray/actor.py [hunk 1]
```diff
@@ -841,7 +841,10 @@ def __getattribute__(self, attr):
         return object.__getattribute__(self, attr)
 
     def __repr__(self):
-        return "Actor(" + self._ray_actor_id.hex() + ")"
+        return "Actor({})".format(self._ray_actor_id)
+
+    def __str__(self):
+        return "RayActor({})".format(self._ray_class_name)
```
Comment (robertnishihara) [inline python/ray/actor.py]: Why put different information in these two functions? I think they should be the same.

File: python/ray/actor.py [hunk 2]
```diff
@@ -841,7 +841,10 @@ def __getattribute__(self, attr):
         return object.__getattribute__(self, attr)
 
     def __repr__(self):
-        return "Actor(" + self._ray_actor_id.hex() + ")"
+        return "Actor({})".format(self._ray_actor_id)
```
Comment (robertnishihara) [inline python/ray/actor.py]: We probably want to keep the `.hex()`, right?

**Review by robertnishihara — COMMENTED** [review]
Good idea!

File: python/ray/actor.py (hunk 1 above)
Comment (richardliaw) [reply python/ray/actor.py]: Hm, not totally sure - I read this particular comment https://stackoverflow.com/a/1436756/2213289 which seems to suggest you want the unique identifier in the repr.

File: python/ray/actor.py (hunk 2 above)
Comment (richardliaw) [reply python/ray/actor.py]: ah fixed

**Comment by robertnishihara** [comment]
Ok, let's remove the `__str__` implementation (it will just use `__repr__`) and put both the class name and ID into the string. How does that sound?

**Comment by richardliaw** [comment]
Would a truncated ID be fine? Like first 6 of the hex? It helps
significantly with readability when debugging.

**Comment by robertnishihara** [comment]
I'd prefer to keep the full ID so that in the future when we have a more fully-fledged UI, people can copy and paste it into the UI and get more information about the actor. The full ID should also help with searching through logs and the GCS and things like that. Also, we use full IDs for all of the other kinds of IDs, so it would be more consistent.

**Comment by richardliaw** [comment]
this is ready for another pass @robertnishihara

**Review by robertnishihara — APPROVED** [review]

**Pull request merged**